fix(ci): aarch64-linux-musl build fails linking aws-lc-sys against glibc symbols

## Problem

The `aarch64-unknown-linux-musl` release build fails during linking because `aws-lc-sys` (pulled in via `reqwest` → `hyper-rustls` → `rustls` → `aws-lc-rs`) is compiled with `aarch64-linux-gnu-gcc` which produces objects referencing glibc-specific symbols that don't exist in musl:

```
undefined reference to `__isoc23_sscanf'
undefined reference to `__fprintf_chk'
undefined reference to `__isoc23_strtol'
undefined reference to `__vsnprintf_chk'
```

### Root cause

The CI step for `aarch64-unknown-linux-musl` installs `gcc-aarch64-linux-gnu` and sets `CARGO_TARGET_AARCH64_UNKNOWN_LINUX_MUSL_LINKER=aarch64-linux-gnu-gcc`. This GNU cross-compiler links against glibc, not musl. When `aws-lc-sys` builds its C code with this toolchain, the resulting `.o` files contain glibc fortified function calls that are absent from musl's libc.

The `x86_64-unknown-linux-musl` target works because `musl-tools` provides a proper musl-targeting `musl-gcc` wrapper on the host architecture.

### Affected workflow

`release-please.yml` → `build` job → `aarch64-unknown-linux-musl` matrix entry.

## Possible solutions

### Option A: Use `cross` (recommended)

Use [`cross`](https://github.com/cross-rs/cross) which provides pre-built Docker images with proper musl cross-compilation toolchains:

```yaml
- name: Install cross
  run: cargo install cross --git https://github.com/cross-rs/cross

- name: Build release binary
  run: cross build --release --target aarch64-unknown-linux-musl -p rinda-cli
```

### Option B: Install a proper aarch64-musl cross toolchain

Install `musl-cross-make` or the `aarch64-linux-musl-cross` toolchain and set `CC_aarch64_unknown_linux_musl` accordingly:

```yaml
- name: Install aarch64 musl cross toolchain
  run: |
    wget -q https://musl.cc/aarch64-linux-musl-cross.tgz
    tar xzf aarch64-linux-musl-cross.tgz -C /opt
    echo "/opt/aarch64-linux-musl-cross/bin" >> $GITHUB_PATH

- name: Build
  run: cargo build --release --target aarch64-unknown-linux-musl -p rinda-cli
  env:
    CARGO_TARGET_AARCH64_UNKNOWN_LINUX_MUSL_LINKER: aarch64-linux-musl-gcc
    CC_aarch64_unknown_linux_musl: aarch64-linux-musl-gcc
```

### Option C: Switch TLS backend to avoid aws-lc-sys

Use `ring` instead of `aws-lc-rs` as the crypto backend, which has better musl cross-compilation support:

```toml
# Cargo.toml
reqwest = { version = "0.13", default-features = false, features = ["json", "rustls-tls-manual-roots", "stream"] }
# or pin rustls to use ring:
rustls = { version = "0.23", default-features = false, features = ["ring", "logging", "std", "tls12"] }
```

## Environment

- Runner: `ubuntu-latest`
- Target: `aarch64-unknown-linux-musl`
- Linker: `aarch64-linux-gnu-gcc` (glibc, not musl)
- Failing crate: `aws-lc-sys v0.38.x`

--- a/.github/workflows/release-please.yml
+++ b/.github/workflows/release-please.yml
@@ -42,6 +42,7 @@ jobs:
             os: ubuntu-latest
             artifact: rinda-cli-linux-arm64
             mcp_artifact: rinda-mcp-linux-arm64
+            use_cross: true
           - target: x86_64-apple-darwin
             os: macos-latest
             artifact: rinda-cli-macos-x64
@@ -74,11 +75,9 @@ jobs:
           sudo apt-get update
           sudo apt-get install -y musl-tools
 
-      - name: Install cross-compilation tools (Linux ARM64 musl)
-        if: matrix.target == 'aarch64-unknown-linux-musl'
-        run: |
-          sudo apt-get update
-          sudo apt-get install -y gcc-aarch64-linux-gnu musl-tools
+      - name: Install cross
+        if: matrix.use_cross
+        run: cargo install cross --git https://github.com/cross-rs/cross
 
       - name: Cache cargo
         uses: Swatinem/rust-cache@v2
@@ -86,14 +85,10 @@ jobs:
           key: ${{ matrix.target }}
 
       - name: Build release binary (rinda-cli)
-        run: cargo build --release --target ${{ matrix.target }} -p rinda-cli
-        env:
-          CARGO_TARGET_AARCH64_UNKNOWN_LINUX_MUSL_LINKER: aarch64-linux-gnu-gcc
+        run: ${{ matrix.use_cross && 'cross' || 'cargo' }} build --release --target ${{ matrix.target }} -p rinda-cli
 
       - name: Build release binary (rinda-mcp)
-        run: cargo build --release --target ${{ matrix.target }} -p rinda-mcp
-        env:
-          CARGO_TARGET_AARCH64_UNKNOWN_LINUX_MUSL_LINKER: aarch64-linux-gnu-gcc
+        run: ${{ matrix.use_cross && 'cross' || 'cargo' }} build --release --target ${{ matrix.target }} -p rinda-mcp
 
       - name: Rename binary (Unix)
         if: runner.os != 'Windows'

